crates/modelardb_compression_python/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+    let timestamps = modelardb_common::array!(uncompressed, 0, TimestampArray);
+    let values = modelardb_common::array!(uncompressed, 1, ValueArray);
+
+    // unwrap() is safe as the timestamps and values are guaranteed to be the same length.
+    let compressed = modelardb_compression::try_compress(
+        univariate_id,
+        timestamps,
+        values,
+        error_bound,
+        &COMPRESSED_SCHEMA,
+    )
+    .unwrap();
+
+    // Convert and return the finished compressed PyArrow record batch.
+    convert::rust_record_batch_to_python(compressed, python)
+}
+
+/// Decompress a [`RecordBatch`] containing compressed segments to one more univariate time series.
+/// An [`ArrowException`] is returned if the schema of `compressed` does not match
+/// [`COMPRESSED_SCHEMA`] or if [`pyo3`] cannot convert the argument from Python to Rust and the
+/// result from Rust to Python.
+#[pyfunction]
+fn decompress<'a>(compressed: &'a PyAny, python: Python<'a>) -> PyResult<&'a PyAny> {
+    // Convert the record batch from Python to Rust and validate the arguments.
+    let compressed = convert::python_record_batch_to_rust(compressed)?;
+    if !fields_has_equal_types(&compressed.schema(), &COMPRESSED_SCHEMA.0) {
+        let error = ArrowError::InvalidArgumentError(
+            "Compressed must only contain compressed segments with metadata and models.".to_owned(),
+        );
+        return Err(convert::to_py_err(error));
+    }
+
+    // Reconstruct the data points from the compressed segments containing metadata and models.
+    modelardb_common::arrays!(
+        compressed,
+        univariate_ids,
+        model_type_ids,
+        start_times,
+        end_times,
+        timestamps,
+        min_values,
+        max_values,
+        values,
+        _error_array
+    );
+
+    let mut univariate_id_builder = UInt64Array::builder(compressed.num_rows());
+    let mut timestamp_builder = TimestampBuilder::with_capacity(compressed.num_rows());
+    let mut value_builder = ValueBuilder::with_capacity(compressed.num_rows());
+
+    for row_index in 0..compressed.num_rows() {
+        models::grid(
+            univariate_ids.value(row_index),
+            model_type_ids.value(row_index),
+            timestamps.value(row_index),
+            start_times.value(row_index),
+            end_times.value(row_index),
+            values.value(row_index),
+            min_values.value(row_index),
+            max_values.value(row_index),
+            &mut univariate_id_builder,
+            &mut timestamp_builder,
+            &mut value_builder,
+        );
+    }
+
+    // Construct, convert, and return the finished PyArrow record batch.
+    let columns: Vec<ArrayRef> = vec![
+        Arc::new(univariate_id_builder.finish()),
+        Arc::new(timestamp_builder.finish()),
+        Arc::new(value_builder.finish()),
+    ];
+
+    // unwrap() is safe as the arrays have the same length and the record batch match the schema.
```

Comment:
Maybe `and the record batch matches the schema` instead.

Reply:
Fixed.